source/hwIo.py

```diff
@@ -178,9 +177,31 @@ def close(self):
 
 	def _notifyReceive(self, data):
 		# Set the timeout for onReceive in case it does a sync read.
-		self._ser.timeout = self._origTimeout
+		self._setTimeout(self._origTimeout)
 		super(Serial, self)._notifyReceive(data)
-		self._ser.timeout = None
+		self._setTimeout(None)
+
+	def _setTimeout(self, timeout):
+		# #6035: pyserial reconfigures all settings of the port when setting a timeout.
+		# This can cause error 'Cannot configure port, some setting was wrong.'
+		# Therefore, manually set the timeouts using the Win32 API.
+		# Adapted from pyserial 3.1.1.
+		timeouts = COMMTIMEOUTS()
+		if timeout is None:
+			pass # default of all zeros is OK
+		elif timeout == 0:
+			timeouts.ReadIntervalTimeout = win32.MAXDWORD
+		else:
+			timeouts.ReadTotalTimeoutConstant = max(int(timeout * 1000), 1)
+		if timeout != 0 and self.ser._interCharTimeout is not None:
+			timeouts.ReadIntervalTimeout = max(int(self.ser._interCharTimeout * 1000), 1)
+		if self.ser._writeTimeout is None:
+			pass
+		elif self.ser._writeTimeout == 0:
+			timeouts.WriteTotalTimeoutConstant = win32.MAXDWORD
+		else:
+			timeouts.WriteTotalTimeoutConstant = max(int(self.ser._writeTimeout * 1000), 1)

```

Comment:
All occurrences of `self.ser` should be `self._ser`. Currently causes `AttributeError`.